### PR TITLE
RSDK-9566: Monitor CPU/Memory usage of viam-server and modular processes with FTDC.

### PR DESCRIPTION
**Pull request by dgottlieb**
In this patch, we just output raw "counters" for things like cpu/system time. I'm going to make a subsequent change to the ftdc parsing to treat those special to generate the actual cpu usage graphs.

### CODE REVIEW

File: ftdc/sys/sys.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+type stats struct {
+	UserCPUSecs     float64
+	SystemCPUSecs   float64
+	ElapsedTimeSecs float64
+	VssMB           float64
+	RssMB           float64
+}
+
+// Stats returns Stats.
+func (sys *UsageStatser) Stats() any {
```
Comment (cheukt) [inline ftdc/sys/sys.go:69]: what does Stats() return on a mac or possibly a windows machine?

**Review by cheukt — COMMENTED** [review]
generally looks good, q around behavior on other systems

File: ftdc/sys/sys.go (hunk 1 above)
Comment (dgottlieb) [reply ftdc/sys/sys.go:69]: I expect the constructor to fail on both -- therefore FTDC would never call `Stats`. But if the constructor succeeds, I'd expect the `Stat` call on line 80 to return an error and we'd just get 0 values out.

File: module/modmanager/manager.go [hunk 2]
```diff
@@ -1143,6 +1151,27 @@ func (m *module) startProcess(
 		return errors.WithMessage(err, "module startup failed")
 	}
 
+	// Turn on process cpu/memory diagnostics for the module process. If there's an error, we should
+	// continue normally, just without FTDC. The control flow for that is a bit messy, so we use a
+	// lambda to leverage early-returns.
+	func() {
```
[the remark this answers is not in the document]
Comment (dgottlieb) [reply module/modmanager/manager.go:1157]: Done

File: ftdc/sys/sys.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+import (
+	"os"
+	"time"
+
+	"github.com/prometheus/procfs"
+)
+
+// On linux, getting the page size is a system call. Cache the page size for the entirety of the
+// progarm lifetime. As opposed to calling it each time we wish to compute the resident memory a
```
[the remark this answers is not in the document]
Comment (dgottlieb) [reply ftdc/sys/sys.go:12]: Done

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**